### ISSUE TITLE
JsonIdentityInfo: Use object equality for specific classes only

### ISSUE THREAD

**Issue by Xfel** [FasterXML/jackson-databind#1486]
Right now, I can set the JsonIdentityInfo serializer to use object identity or equality using `SerializationConfig.USE_EQUALITY_FOR_OBJECT_ID`, but this will change the setting for the whole ObjectWriter. It is not possible to modify this for specific types only.

**Comment by cowtowncoder** [comment]
Makes sense. I wonder if this could be implemented through new (in 2.8) "config overrides" functionality, since that is based on per-property settings.

**Comment by cowtowncoder** [comment]
No plans to implement; closing as part of bigger issue clean up.

May be re-opened if someone is to work on the issue.
